core/src/main/java/com/klaytn/caver/wallet/keyring/KeyStore.java

```diff
@@ -0,0 +1,1160 @@
+            byte[] result = new byte[16 + cipherText.length];
+
+            System.arraycopy(derivedKey, 16, result, 0, 16);
+            System.arraycopy(cipherText, 0, result, 16, cipherText.length);
+
+            return Hash.sha3(result);
+        }
+
+        /**
+         * Getter function for cipher.
+         * @return String
+         */
+        public String getCipher() {
+            return cipher;
+        }
+
+        /**
+         * Setter function for cipher.
+         * @param cipher cipher algorithm name.
+         */
+        public void setCipher(String cipher) {
+            this.cipher = cipher;
+        }
+
+        /**
+         * Getter function for cipher text.
+         * @return String
+         */
+        public String getCiphertext() {
+            return ciphertext;
+        }
+
+        /**
+         * Setter function for cipher text
+         * @param ciphertext encrypted key
+         */
+        public void setCiphertext(String ciphertext) {
+            this.ciphertext = ciphertext;
+        }
+
+        /**
+         * Getter function for CipherParams.
+         * @return KeyStore.CipherParams.
+         */
+        public KeyStore.CipherParams getCipherparams() {
+            return cipherparams;
+        }
+
+        /**
+         * Setter function for CipherParams.
+         * @param cipherparams KeyStore.CipherParams.
+         */
+        public void setCipherparams(KeyStore.CipherParams cipherparams) {
+            this.cipherparams = cipherparams;
+        }
+
+        /**
+         * Getter function for kdf name.
+         * @return String
+         */
+        public String getKdf() {
+            return kdf;
+        }
+
+        /**
+         * Setter function for kdf name.
+         * @param kdf kdf name.
+         */
+        public void setKdf(String kdf) {
+            this.kdf = kdf;
+        }
+
+        /**
+         * Getter function for KdfParams.
+         * @return KeyStore.KdfParams
+         */
+        public KeyStore.KdfParams getKdfparams() {
+            return kdfparams;
+        }
+
+        /**
+         * Setter function for KdfParams.
+         * @param kdfparams KeyStore.KdfParams.
+         */
+        @JsonTypeInfo(
+                use = JsonTypeInfo.Id.NAME,
+                include = JsonTypeInfo.As.EXTERNAL_PROPERTY,
+                property = "kdf")
+        @JsonSubTypes({
+                @JsonSubTypes.Type(value = Pbkdf2KdfParams.class, name = Pbkdf2KdfParams.NAME),
+                @JsonSubTypes.Type(value = KeyStore.ScryptKdfParams.class, name = ScryptKdfParams.NAME)
+        })
+        // To support my Ether Wallet keys uncomment this annotation & comment out the above
+        //  @JsonDeserialize(using = KdfParamsDeserialiser.class)
+        // Also add the following to the ObjectMapperFactory
+        // objectMapper.configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
```

Comment:
Need this comment?

Reply:
This KeyStore class copied to WalletFile.java.
This comment is redundant. However, I left it in case I don't know in the future.